src/Swashbuckle.AspNetCore.SwaggerUI/SwaggerUIBuilderExtensions.cs

```diff
@@ -38,10 +43,38 @@ public static IApplicationBuilder UseSwaggerUI(this IApplicationBuilder app, Swa
             if (options.ConfigObject.Urls == null)
             {
                 var hostingEnv = app.ApplicationServices.GetRequiredService<IWebHostEnvironment>();
-                options.ConfigObject.Urls = new[] { new UrlDescriptor { Name = $"{hostingEnv.ApplicationName} v1", Url = "v1/swagger.json" } };
+                options.ConfigObject.Urls = new[]
+                    { new UrlDescriptor { Name = $"{hostingEnv.ApplicationName} v1", Url = "v1/swagger.json" } };
             }
 
             return app.UseSwaggerUI(options);
         }
+
+
```

Comment:
The implementation here looks a little different to how it's been done for the [SwaggerMiddleware](https://github.com/domaindrivendev/Swashbuckle.AspNetCore/blob/master/src/Swashbuckle.AspNetCore.Swagger/DependencyInjection/SwaggerBuilderExtensions.cs#L42), which I know has gone through a couple of iterations. Could this be implemented in the same way or is there a good reason for them to be different?

Reply:
When the convention builder is returned, you are then able to further use it and add Authentication to it. Or what do you mean

Reply:
Returning the builder makes sense. I’m referring to the rest of the method impl. Just do a diff to see what I mean

Reply:
Now I see what you mean. I will adjust it. The difference is that we try to get the settings from DI when the action is null

Reply:
Ok I got back to it and had a look at it. 

We need this implementation here. The problem is that otherwise the static files will produce an 401 when you try to get them. (This is prevented by the .Use(....)

If you like I can remove the else branch for getting the settings from DI. --> setupAction must then be provided for this method.